ibm/service/appconfiguration/data_source_ibm_app_config_collection.go

```diff
@@ -0,0 +1,234 @@
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Collection description.",
+			},
+			"tags": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Tags associated with the collection.",
+			},
+			"created_time": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Creation time of the collection.",
+			},
+			"updated_time": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Last modified time of the collection data.",
+			},
+			"href": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Collection URL.",
+			},
+			"features_count": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Number of features associated with the collection.",
+			},
+			"properties_count": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Number of properties associated with the collection.",
+			},
+			"features": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of Features associated with the collection.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"feature_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "feature id.",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Name of the Feature.",
+						},
+					},
+				},
+			},
+			"properties": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of properties associated with the collection.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"property_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "property id.",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Name of the Property.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIbmAppConfigCollectionRead(d *schema.ResourceData, meta interface{}) error {
+	guid := d.Get("guid").(string)
+
+	appconfigClient, err := getAppConfigClient(meta, guid)
+	if err != nil {
+		return fmt.Errorf("getAppConfigClient failed %s", err)
+	}
+
+	options := &appconfigurationv1.GetCollectionOptions{}
+
+	options.SetCollectionID(d.Get("collection_id").(string))
+
+	if _, ok := d.GetOk("include"); ok {
+		includes := []string{}
+		for _, includeItem := range d.Get("include").([]interface{}) {
+			includes = append(includes, includeItem.(string))
+		}
+		options.SetInclude(includes)
+	}
+	if _, ok := d.GetOk("expand"); ok {
+		options.SetExpand(d.Get("expand").(bool))
+	}
+
+	result, response, err := appconfigClient.GetCollection(options)
+	if err != nil {
+		log.Printf("[DEBUG] GetCollection failed %s\n%s", err, response)
```

Comment:
You can remove this kind of log stmts since we already giving back complete response

Reply:
Addressed it